### PR TITLE
update SMSG_GARRISON_ADD_MISSION_RESULT

### PR DESCRIPTION
**Pull request by Traesh**

### CODE REVIEW

**Comment by soulfrost** [comment]
@Traesh @mdX7 , u are wrong, this called "overmaxRewards" in client

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 1]
```diff
@@ -192,5 +195,26 @@ public static void HandleGetGarrisonInfoResult(Packet packet)
...
+            var rewardsCount = packet.ReadUInt32("RewardsCount");
+            var bonusRewardsCount = packet.ReadUInt32("BonusRewardsCount");
+            
+            for (int i = 0; i < rewardsCount; i++)
+                ReadGarrisonMissionRewards(packet, "MissionRewards", i);
+            
+            for (int i = 0; i < bonusRewardsCount; i++)
+                ReadGarrisonMissionRewards(packet, "MissionBonusRewards", i);
+
+            packet.ReadBit("Success");
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: resetBitReader should be before u start reading bit/bits - not after them

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 2]
```diff
@@ -25,15 +25,18 @@ public static void ReadGarrisonMissionOvermaxRewards(Packet packet, params objec
...
+
+        public static void ReadGarrisonMissionRewards(Packet packet, params object[] indexes)
+        {
+            packet.ReadInt32<ItemId>("ItemID", indexes);
+            packet.ReadUInt32("Quantity", indexes);
+            packet.ReadInt32("CurrencyID", indexes);
+            packet.ReadUInt32("CurrencyQuantity", indexes);
+            packet.ReadUInt32("FollowerXP", indexes);
+            packet.ReadUInt32("BonusAbilityID", indexes);
+            packet.ReadInt32("Unknown", indexes);
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: cant find name in client, but this is `ItemIconFileDataId` or called like this

File: WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs [hunk 3]
```diff
@@ -192,5 +195,26 @@ public static void HandleGetGarrisonInfoResult(Packet packet)
                         ReadGarrisonFollower(packet, "Follower", i, j);
             }
         }
+
+        [Parser(Opcode.SMSG_GARRISON_ADD_MISSION_RESULT)]
+        public static void HandleGarrisonAddMissionResult(Packet packet)
+        {
+            packet.ReadInt32("GarrTypeID");
+            packet.ReadInt32("Result");
```
Comment (soulfrost) [inline WowPacketParserModule.V7_0_3_22248/Parsers/GarrisonHandler.cs]: make enum from core with GarrisonError ? same for GarrTypeId

**Comment by mdX7** [comment]
I agree, seems like I checked WoD client instead of Legion/BfA. Checked latest client (8.0.1.27178), its called overmax in there.

**Pull request closed without merging**